//package ;

message BaseAttr
{

  enum ePart
  {
    kPart_Null = 0;
    kPart_Fighter = 1;
    kPart_Armor = 2;
    kPart_Wingman = 3;
    kPart_Attache = 4;//附属
  }

  enum eQuality
  {
    kQuality_Gray = 0;
    kQuality_White = 1;
    kQuality_Green= 2;
    kQuality_Blue = 3;
    kQuality_Purple = 4;
    kQuality_Orange = 5;
  }

  enum eQualityRank
  {
    kQuaRank_0 = 0;
    kQuaRank_1 = 1;
    kQuaRank_2 = 2;
    kQuaRank_Max = 3;
    kQuaRank_Gold = 4;
  }

    required ePart part = 1[default = kPart_Null];
    required int32 level = 2[default = 1];
    required eQuality quality = 3[default = kQuality_White];
    required eQualityRank quaRank = 4[default = kQuaRank_0];

    optional int32 hp = 5;
    optional int32 defence = 6;
    optional int32 atk = 7;
    optional int32 atk_speed = 8;
}